reports/cost_variation.go

```diff
@@ -0,0 +1,251 @@
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package reports
+
+import (
+"context"
+"database/sql"
+"fmt"
+"strconv"
+"strings"
+"time"
+
+"github.com/360EntSecGroup-Skylar/excelize"
+"github.com/trackit/jsonlog"
+
+"github.com/trackit/trackit-server/aws"
+"github.com/trackit/trackit-server/aws/usageReports/history"
+"github.com/trackit/trackit-server/costs/diff"
+)
+
+type costVariationProduct map[time.Time]diff.PricePoint
+type costVariationReport map[string]costVariationProduct
+
+const (
+	freqDaily = iota
+	freqMonthly
+)
+
+const costVariationLastMonthSheetName = "Cost Variations (Last Month)"
+const costVariationLast6MonthsSheetName = "Cost Variations (Last 6 Months)"
+
+var costVariationLastMonth = module {
+	Name:          "Cost Variations (Last Month)",
+	SheetName:     costVariationLastMonthSheetName,
+	ErrorName:     "costVariationLastMonthError",
+	GenerateSheet: costVariationGenerateLastMonth,
+}
+
+var costVariationLast6Months = module {
+	Name:          "Cost Variations (Last 6 Months)",
+	SheetName:     costVariationLast6MonthsSheetName,
+	ErrorName:     "costVariationLastSixMonthsError",
+	GenerateSheet: costVariationGenerateLast6Months,
+}
+
+func costVariationGenerateLastMonth(ctx context.Context, aas []aws.AwsAccount, date time.Time, _ *sql.Tx, file *excelize.File) (err error) {
+	var dateRange = make([]time.Time, 2)
+	if date.IsZero() {
+		dateRange[0], dateRange[1] = history.GetHistoryDate()
+	} else {
+		dateRange[0] = date
+		dateRange[1] = time.Date(date.Year(), date.Month()+1, 0, 23, 59, 59, 999999999, date.Location()).UTC()
+	}
+	return costVariationGenerateSheet(ctx, aas, dateRange, freqDaily, file)
+}
+
+func costVariationGenerateLast6Months(ctx context.Context, aas []aws.AwsAccount, date time.Time, _ *sql.Tx, file *excelize.File) (err error) {
+	var dateRange = make([]time.Time, 2)
+	if date.IsZero() {
+		_, dateRange[1] = history.GetHistoryDate()
+	} else {
+		dateRange[1] = time.Date(date.Year(), date.Month() + 1, 0, 23, 59, 59, 999999999, date.Location()).UTC()
+	}
+	dateRange[0] = time.Date(dateRange[1].Year(), dateRange[1].Month() - 5, 1, 0, 0, 0, 0, dateRange[1].Location()).UTC()
+
+	return costVariationGenerateSheet(ctx, aas, dateRange, freqMonthly, file)
+}
+
+func costVariationGenerateSheet(ctx context.Context, aas []aws.AwsAccount, dateRange []time.Time, frequency int, file *excelize.File) (err error) {
+	data, err := costVariationGetData(ctx, aas, dateRange, frequency)
+	if err == nil {
+		return costVariationInsertDataInSheet(ctx, aas, dateRange, frequency, file, data)
+	}
+	return
+}
+
+func costVariationGetData(ctx context.Context, aas []aws.AwsAccount, dateRange []time.Time, frequency int) (data map[aws.AwsAccount]costVariationReport, err error){
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	var aggregation string
+	if frequency == freqMonthly {
```

Comment:
Use a slice to transform the iota to a string (see the comment at the top of the file)